### PR TITLE
config data: add .get_unquoted() convenience method to get quoted string unquoted

### PR DESCRIPTION
**Pull request by t-chaik**

### CODE REVIEW

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -216,6 +217,20 @@ def get_method(self, args, kwargs):
...
+            raise InterpreterException('Get method takes one or two arguments.')
+        name = args[0]
+        if name in self.held_object:
+            val = self.held_object.get(name)[0]
+        elif len(args) > 1:
+            val = args[1]
+        else:
+            raise InterpreterException('Entry %s not in configuration data.' % name)
+        if val[0] == '"' and val[-1] == '"':
+          return val[1:len(val)-1]
```
Comment (ignatenkobrain) [inline mesonbuild/interpreter.py]: __[flake8]__
* [E111] indentation is not a multiple of four
* [E226] missing whitespace around arithmetic operator

**Comment by jpakkane** [comment]
Test failure seems relevant:

    c:\projects\meson\b ux3aurv_\config3.h(33) : error C2295: escaped '"' : is illegal in macro definition

**Comment by jpakkane** [comment]
⚔️ conflicts. This should also get a mention in the release notes and have the docs updated.

**Comment by jpakkane** [comment]
Thanks, and sorry for the delay.

File: docs/markdown/Reference-manual.md [hunk 2]
```diff
@@ -1522,6 +1522,11 @@ page](Configuration.md) It has three methods:
   value has not been set returns `default_value` if it is defined
   *(added 0.38.0)* and errors out if not
 
+- `get_unquoted(varname, default_value)` returns the value of `varname`
+  but without surrounding double quotes (`"`). If the value has not been
+  set returns `default_value` if it is defined and errors out if not.
+  Available since 0.43.0
```
Comment (jon-turney) [inline docs/markdown/Reference-manual.md]: It looks like this wasn't actually merged until after 0.43.0, and so first makes an appearance in 0.44.0

**Pull request merged**